### PR TITLE
checkmemberobjects apis

### PR DESCRIPTION
**Pull request by davidmu1**

### CODE REVIEW

File: api-reference/beta/api/directoryobject-checkmemberobjects.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+| Delegated (work or school account)     | Directory.Read.All, Directory.ReadWrite.All |
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | Directory.Read.All, Directory.ReadWrite.All |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+POST /me/manager/checkMemberObjects
```
Comment (dkershaw10) [inline api-reference/beta/api/directoryobject-checkmemberobjects.md]: Are you going to have the full list here - like also group, servicePrincipal and device?
Also is `POST ../directoryObjects/{id}/checkMemberObjects` possible?

File: api-reference/beta/api/directoryrole-checkmemberobjects.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+title: "directoryRole: checkMemberObjects"
```
Comment (dkershaw10) [inline api-reference/beta/api/directoryrole-checkmemberobjects.md]: Remove this file.  I don't think directoryRoles can be members of anything.  They aren't security principals.  Pretty sure this doesn't work.

File: api-reference/beta/api/device-checkmemberobjects.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+author: "davidmu1"
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# device: checkMemberObjects
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Check for membership in a list of groups, directory roles, or administrative units for the specified device object. This function is transitive.
```
Comment (dkershaw10) [inline api-reference/beta/api/device-checkmemberobjects.md]: It's not a "function"  so we shouldn't call it that - strictly it's a service action.
That said, I would say "This method is transitive." or "This API is transitive."

File: api-reference/beta/api/user-checkmemberobjects.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Check for membership in a list of group, directory role, or administrative unit objects for the specified user object. This function is transitive.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
```
Comment (dkershaw10) [inline api-reference/beta/api/user-checkmemberobjects.md]: Not convinced this is correct...

File: api-reference/beta/resources/directoryrole.md [hunk 5]
```diff
@@ -30,6 +30,7 @@ This resource supports:
 |[Remove a member](../api/directoryrole-delete-member.md) |[directoryObject](directoryobject.md)| Remove a user from the directory role.|
 |[List scoped-role members](../api/directoryrole-list-members.md) |[scopedRoleMembership](scopedrolemembership.md) collection| List the members of this directory role that are scoped to [administrative units](administrativeunit.md), through the scopedRoleMembership resource collection.|
 |[delta](../api/directoryrole-delta.md)|directoryRole collection| Get incremental changes for directory roles. |
+|[checkMemberObjects](../api/directoryrole-checkmemberobjects.md) | String collection | Check for membership in a list of group, directory role, or administrative unit objects. |
```
Comment (dkershaw10) [inline api-reference/beta/resources/directoryrole.md]: Remove this change

File: concepts/changelog.md [hunk 6]
```diff
@@ -77,6 +77,12 @@ For details about known issues with Microsoft Graph APIs, see [Known issues](kno
...
 | Addition | Beta | Added these operations to support the unifiedRoleAssignment resource: <br><ul><li>[List unifiedRoleAssignments](/graph/api/rbacapplication-list-roleassignments?view=graph-rest-beta) - Get a list of unifiedRoleAssignment objects for the provider.</li><li>[Create unifiedRoleAssignment]() - Create a new unifiedRoleAssignment object.</li><li>[Get unifiedRoleAssignment](/graph/api/unifiedroleassignment-get?view=graph-rest-beta) - Retrieve the properties and relationships of a unifiedRoleAssignment object.</li><li>[Delete unifiedRoleAssignment](/graph/api/unifiedroleassignment-delete?view=graph-rest-beta) - Delete a unifiedRoleAssignment object. |
 | Addition | Beta | Added the [List accessReviews](/graph/api/accessreview-list?view=graph-rest-beta) operation. |
+| Addition | Beta | Added the [checkMemberObjects](/graph/api/device-checkmemberobjects?view=graph-rest-beta) operation for the [device](/graph/api/resources/device?view=graph-rest-beta) resource. |
```
Comment (dkershaw10) [inline concepts/changelog.md]: What about TOC updates?

**Comment by davidmu1** [comment]
#sign-off

File: api-reference/beta/api/device-checkmemberobjects.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    "ac38546e-ddf3-437a-ac5c-27a94cd7a0f1"
+  ]
+}
+```
+
+### Response
+
+The following is an example of the response.
+
+> [!NOTE]
```
Comment (Lauragra) [inline api-reference/beta/api/device-checkmemberobjects.md]: For this boilerplate text, we use the less noticeable format for Notes, so it's just the ">".

File: api-reference/beta/api/device-checkmemberobjects.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+|:-------------|:------------|:------------|
+| ids | String collection | A collection that contains the object IDs of the groups, directory roles, administrative units, or roleTemplate IDs of directory roles, in which to check membership. Up to 20 objects may be specified. |
+
+## Response
+
+If successful, this method returns a `200 OK` response code and a String collection object in the response body.
+
+## Examples
+
+The following is an example of how to call this API.
```
Comment (Lauragra) [inline api-reference/beta/api/device-checkmemberobjects.md]: Global: Revise as "The following example shows how to call this API."

Comment (davidmu1) [reply api-reference/beta/api/device-checkmemberobjects.md]: Changed.

File: api-reference/beta/api/device-checkmemberobjects.md (hunk 7 above)
Comment (davidmu1) [reply api-reference/beta/api/device-checkmemberobjects.md]: Changed.

File: api-reference/beta/api/user-checkmemberobjects.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+Check for membership in a list of group, directory role, or administrative unit objects for the specified user object. This method is transitive.
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+| Permission type                        | Permissions (from least to most privileged) |
+|:---------------------------------------|:--------------------------------------------|
+| Delegated (work or school account)     | User.ReadBasic, User.Read.All, User.ReadWrite.All<br><br>In addition:<br><ul><li>If checking for membership in groups: Group.Read.All, Group.ReadWrite.All</li><li>If checking for membership in administrative units: AdministrativeUnit.Read.All, AdministrativeUnit.ReadWrite.All</li><li>If checking for membership in directory roles: RoleManagement.Read.Directory, RoleManagement.ReadWrite.Directory</li></ul>Directory.Read.All, Directory.ReadWrite.All, Directory.AccessAsUser.All  |
```
Comment (Lauragra) [inline api-reference/beta/api/user-checkmemberobjects.md]: I tried to make the text here more concise. If it looks okay, please apply the same to the other occurrences.

File: api-reference/beta/api/serviceprincipal-checkmemberobjects.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    "62e90394-69f5-4237-9190-012177145e10",
+    "86a64f51-3a64-4cc6-a8c8-6b8f000c0f52",
+    "ac38546e-ddf3-437a-ac5c-27a94cd7a0f1"
+  ]
+}
+```
+
+### Response
+
+The following is an example of the response. Note: The response object shown here might be shortened for readability. All the properties will be returned from an actual call.
```
Comment (Lauragra) [inline api-reference/beta/api/serviceprincipal-checkmemberobjects.md]: The note text should be on a separate line with the alert notation, like this:

>**Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

File: concepts/changelog.md (hunk 6 above)
Comment (Lauragra) [reply concepts/changelog.md]: For any method that we list in a methods table, we should also have it in the TOC, in the same order, FYI. If some methods are missing from the TOC, we should look into that. Complex types are not included in the TOC.

File: api-reference/beta/api/serviceprincipal-checkmemberobjects.md (hunk 10 above)
Comment (davidmu1) [reply api-reference/beta/api/serviceprincipal-checkmemberobjects.md]: Changed.

File: api-reference/beta/api/user-checkmemberobjects.md (hunk 9 above)
Comment (davidmu1) [reply api-reference/beta/api/user-checkmemberobjects.md]: Changed.

**Comment by davidmu1** [comment]
Found out the APIs aren't live yet, need to wait until they are.

**Comment by davidmu1** [comment]
#sign-off

File: concepts/changelog.md (hunk 6 above)
Comment (Lauragra) [reply concepts/changelog.md]: @davidmu1 , are you planning to add the user- and group- function topics to the TOC as well?

**Comment by CelesteDG** [comment]
@davidmu1 - Any concerns with getting master merged into this branch to get it up-to-date?

**Comment by Lauragra** [comment]
@CelesteDG , I do that before I merge any PRs, FYI.

**Comment by CelesteDG** [comment]
Thank you @Lauragra - This PR is ready to merge to master. APIs are live.

**Comment by Lauragra** [comment]
@CelesteDG , yep, I'm waiting on a response from @davidmu1 . We're inconsistent in how we're surfacing the new APIs in the TOC in the PR.

**Comment by davidmu1** [comment]
@Lauragra, @CelesteDG - Laura, maybe you can explain what the rules are for adding TOC entries. I did not add TOC entries for Users or Groups because most of the other methods for these resources are not listed in the TOC, only on the resource page.

**Comment by davidmu1** [comment]
@Lauragra, @CelesteDG, @dkershaw10 - Laura, after discussing with Dan, we would like to let this PR go through as is and cover correcting the TOC for Users and Groups in another PR. I have created workitem 810673 to cover the work, which will likely be prioritized for the next sprint. Thanks.

**Pull request merged**